main/EventSystem.cpp

```diff
@@ -2442,27 +2440,28 @@ void CEventSystem::ExportDomoticzDataToLua(lua_State *lua_state, uint64_t device
 			_hardwareNames[ID] = tlist;
 		}
 	}
+	
+	Json::Value tempjson;
+	m_webservers.GetJSonDevices(tempjson, "true", "", "dzvID", "", "", "", true, false, false, 0, "");
```

Comment:
This is a very very expensive call... this takes a lot of CPU... specially with systems with lots of devices (like mine)
It seems you are now calling this function for each lua evaluation ?
Why not use the device states that are stored after each update ? That's why it was created... and it should always be up2date